### PR TITLE
chore(observability): Namespace enterprise metrics with `vector`

### PR DESCRIPTION
**Pull request by 001wwang**
Closes [OP-452](https://datadoghq.atlassian.net/browse/OP-452)

This PR
- Namespaces enterprise `internal_metrics` with `vector` and `host_metrics` with `vector.host`
- Preserves a `pipelines` namespace metric

## Testing

Build this branch locally and create a new OP configuration in Datadog. The below examples use the following configuration

```toml
[api]
enabled = true

[enterprise]
...

[sources.foo_op_452]
type = "demo_logs"
format = "json"

[sources.stdin_op_452]
type = "stdin"

[transforms.remap_op_452]
type = "remap"
inputs = ["foo_op_452"]
source = '''
  log("I'm an internal log")
'''

[sinks.bar_op_452]
type = "blackhole"
inputs = ["*"]
```

Run your configuration `target/debug/vector -c {your config}`, and navigate to DD metrics explorer. You should see metrics like the following
- `vector` namespaced metrics
Screen Shot 2022-06-08 at 8 07 42 AM

- `vector.host` namespaced host metrics

Screen Shot 2022-06-08 at 8 07 57 AM

- `pipelines.component_received_bytes_total`

Screen Shot 2022-06-08 at 8 08 27 AM

### CODE REVIEW

**Comment by binarylogic** [comment]
Very nice! Thank you for doing this so quickly.

**Review by binarylogic — APPROVED** [review]

**Review by leebenson — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

File: src/config/enterprise.rs [hunk 1]
```diff
@@ -505,6 +516,27 @@ fn setup_metrics_reporting(
...
+    let pipelines_namespace_metrics = RemapConfig {
+        source: Some(r#".namespace = "pipelines""#.to_string()),
+        ..Default::default()
+    };
+
+    let metering_metrics = RemapConfig {
+        source: Some(
+            r#"
+            .namespace = "datadog.observability_pipelines"
+            .name = "billed_bytes_total"
```
Comment (jszwedko) [inline src/config/enterprise.rs:533]: I'd suggest `ingested_bytes_total` here to match the other billing metrics on https://docs.datadoghq.com/account_management/billing/usage_metrics/

**Review by jszwedko — CHANGES_REQUESTED** [review]
Just one comment about the billing metric name, otherwise 👍

File: src/config/enterprise.rs (hunk 1 above)
Comment (001wwang) [reply src/config/enterprise.rs:533]: Ah, yes, that sounds good to me. cc: @binarylogic

Comment (jszwedko) [reply src/config/enterprise.rs:533]: Or, I guess, `datadog.estimated_usage.observability_pipelines.ingested_bytes` to most closely match.

Comment (binarylogic) [reply src/config/enterprise.rs:533]: Agree! I did not see these.

Comment (001wwang) [reply src/config/enterprise.rs:533]: Just for a sanity check, I'll check in with Revenue Engineering on naming best practices.

**Review by jszwedko — APPROVED** [review]

File: src/config/enterprise.rs (hunk 1 above)
Comment (001wwang) [reply src/config/enterprise.rs]: As per our discussion offline, we'll leave this out for now.

**Pull request merged**